core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -1124,7 +1124,8 @@ public static String replace(
     NEOVIEW("Neoview", null, NullCollation.HIGH),
     SYBASE("Sybase", null, NullCollation.HIGH),
     TERADATA("Teradata", "\"", NullCollation.HIGH),
-    HSQLDB("Hsqldb", null, NullCollation.HIGH),
+    //Source: Table 13.27 at http://hsqldb.org/doc/2.0/guide/dbproperties-chapt.html
+    HSQLDB("Hsqldb", null, NullCollation.FIRST),
```

Comment:
from what I remember this info is also pulled from the db on initial connection. Could you confirm this is what is being pulled from our hsqldb?

Reply:
Currently even if requested from the DB it doesn't get applied to the dialect. Default null collation applies.